### PR TITLE
Add ui tests for functions

### PR DESCRIPTION
**Pull request by obeis**
Updates #99

cc @oli-obk

### CODE REVIEW

File: tests/ui/fn/ok.stderr [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+       
+               fn is_true(v: bool) -> bool {}
+       
+               fn add(x: i64, y: i64) -> i64 {}
+       
+               fn return_multi(a: i32, b: i32) -> (i32, i32) {}
+           }
+       ]
+       
+    1: failed to parse [
```
Comment (compiler-errors) [inline tests/ui/fn/ok.stderr]: This is a parser error. Have you looked at the other tests that demonstrate the difference in a-mir-formality syntax?

File: tests/ui/fn/ok.🔬 [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[
+    crate Foo {
+        fn no_args() {}
+
+        fn with_return() -> &str {}
```
Comment (compiler-errors) [inline tests/ui/fn/ok.🔬]: For the record, `with_return` isn't valid Rust:

```
error[E0106]: missing lifetime specifier
 --> src/lib.rs:3:29
  |
3 |         fn with_return() -> &str {}
  |                             ^ expected named lifetime parameter
  |
  = help: this function's return type contains a borrowed value, but there is no value for it to be borrowed from
help: consider using the `'static` lifetime
  |
3 |         fn with_return() -> &'static str {}
```

File: tests/ui/fn/ok.🔬 [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[
+    crate Foo {
+        fn no_args() {}
```
Comment (oli-obk) [inline tests/ui/fn/ok.🔬]:
```suggestion
        fn no_args<>() {}
```

I'd guess you need sth like this, but not sure how formality parses functions

File: tests/ui/fn/ok.🔬 [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+        fn with_return() -> String {}
```
Comment (oli-obk) [inline tests/ui/fn/ok.🔬]: Formality won't know this type, it has no standard library

File: tests/ui/fn/ok.🔬 [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+        fn is_true(v: bool) -> bool {}
+
+        fn add(x: i64, y: i64) -> i64 {}
+
+        fn return_multi(a: i32, b: i32) -> (i32, i32) {}
```
Comment (oli-obk) [inline tests/ui/fn/ok.🔬]: Not sure formality knows tuples

**Review by oli-obk — CHANGES_REQUESTED** [review]

**Comment by oli-obk** [comment]
I haven't forgotten about this 😆

File: tests/ui/fn/ok.stderr [hunk 6]
```diff
@@ -0,0 +1 @@
+Error: failed to prove {@ wf((rigid &(shared) !lt_2 !ty_1))} given {}, got {}
```
Comment (oli-obk) [inline tests/ui/fn/ok.stderr]: about which function is this error?

maybe create one test per failing function.

File: tests/ui/fn/ok.🔬 [hunk 7]
```diff
@@ -0,0 +1,10 @@
+[
+    crate Foo {
+        fn no_args<>() -> () where [] { trusted }
+        fn one_args<ty T, lt l>(&l T) -> () where [] { trusted }
+        fn one_return<ty T, lt l, ty U>(&l T) -> U where [] { trusted }
```
Comment (oli-obk) [inline tests/ui/fn/ok.🔬]: Leave a comment explaining what Rust signature this signature is equivalent to. this is `fn foo<'a, T, U>(_: &'a T) -> U`, and previously you had different functions.

Also leave a comment explaining what the tests are testing, as I can't tell from the test.

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**